src/biogeochem/FATESFireNoDataMod.F90

```diff
@@ -0,0 +1,177 @@
+module FATESFireNoDataMod
+
+#include "shr_assert.h"
+
+  !-----------------------------------------------------------------------
+  ! !DESCRIPTION:
+  ! module for FATES when not obtaining fire inputs from data
+  !
+  ! !USES:
+  use shr_kind_mod, only: r8 => shr_kind_r8, CL => shr_kind_CL
+  use shr_log_mod, only: errmsg => shr_log_errMsg
+  use abortutils, only: endrun
+  use clm_varctl, only: iulog
+  use decompMod, only: bounds_type
+  use CNFireMethodMod, only: cnfire_method_type
+  use CNFireBaseMod, only: cnfire_base_type
+  !
+  implicit none
+  private
+  !
+  ! !PUBLIC TYPES:
+  public :: fates_fire_no_data_type
+  !
+  type, extends(cnfire_base_type) :: fates_fire_no_data_type
+      ! !PRIVATE MEMBER DATA:
+      real(r8), public, pointer :: lnfm24(:)  ! Daily avg lightning by grid cell (#/km2/hr)
```

Comment:
Let me explain my thought process. The NoData method has two options:
1) use_fates_spitfire = 0  i.e. no fire
2) use_fates_spitfire = 1  i.e. fire with no external inputs for lightning etc.

In case (2) the model uses `EDParamsMod, only: ED_val_nignitions`. Although this is a single value of global lightning, I decided to run it through the same steps as the 3D lightning and end up with lnfm24, to allow users the flexibility of calculating time-varying lightning in the code.

If everyone is opposed to this logic, I could skip the lnfm24 calculations in the NoData method and place `ED_val_nignitions` directly in the calculation of lightning24 in `clmfates_interfaceMod.F90`.

Reply:
OK I see. That does make the calculation more expensive because it's over an array rather than a single value. I see the point about making the code in FATES identical though. But, I don't think keeping it as an array here really helps to compare the two options. What would be useful is to make sure the constant in FATES is in same units as the lightning data here. Then you can just visually compare the lightning data with that single value. And the lightning data map can help guide you as what the single constant should be. If lnfm24 is being written out to the history file I could see how it could be useful to compare the two methods that way. But, still one output is a constant the the other varying in space and time, so you really just need to know the value of the constant to do that comparison.

So I would vote to remove the 3D data here, and just use a single constant. Just make sure the single constant is in the same units as lnfm24 so that you can easily compare them.

Reply:
It would be useful to have @jkshuman comment on this. 
I'm not sure I understand the alternatives. 
It seems that someone could create a global constant value in the format of the 3D lightening if they wanted to run it through that way.

Reply:
That's a good point @lmkueppers. That's a totally valid and easy thing to do, and you could do that to show that you get the same results with either method without having to keep the 3D data around. So yeah, I vote to remove it from here.

Reply:
I added some discussion of this point to the FATES side. I think that's the crux of how this should be handled. Once that's determined it will likely dictate the solution here. So let's see what FATES developers want to see on that side.

Reply:
I agree with removing the 3D. @ekluzek also makes a good point about printing out the lightning data.

Reply:
@ekluzek wrote:
> You don't need the lightning data for the noData method.

Based on the conversation this far, I have gone ahead and removed `lnfm24` from the NoData case and made the corresponding accumulation procedures empty.

Our next choice is this. I vote for option (1) and explain my reasoning below:
1) An if statement in `clmfates_interfaceMod.F90`
```suggestion
if (use_fates_spitfire > 1) then
   this%fates(nc)%bc_in(s)%lightning24 = cnfire_method_inst%lnfm24(g) * 24._r8  ! #/km2/hr to #/km2/day
else
   this%fates(nc)%bc_in(s)%lightning24 = ED_val_nignitions / days_per_year  ! #/km2/yr to #/km2/day
end if```
with no if statement in `SFMainMod.F90` (SPITFIRE)
```currentSite%NF = bc_in%lightning24 * cg_strikes```

OR
2) An if statement in each, as follows:
```suggestion
if (use_fates_spitfire > 1) then
   this%fates(nc)%bc_in(s)%lightning24 = cnfire_method_inst%lnfm24(g)
else
   this%fates(nc)%bc_in(s)%lightning24 = 0._r8
end if```

```suggestion
if (use_fates_spitfire > 1) then
   currentSite%NF = bc_in%lightning24 * 24._r8 * cg_strikes  ! #/km2/hr to #/km2/day
else
   currentSite%NF = ED_val_nignitions * years_per_day * cg_strikes  ! #/km2/yr to #/km2/day
end if```

I vote for (1) because:
- SPITFIRE's code becomes simpler with no if statement and no units conversions
- `clmfates_interfaceMod.F90` converts units and passes ignitions to FATES under all circumstances, i.e. we have uniform treatment of all ignition options

My next commit will include option (1). Of course I am happy to change to option (2) if people prefer that.

Reply:
> I agree with removing the 3D. @ekluzek also makes a good point about printing out the lightning data.

3D removed.

Regarding sending the variable lnfm to history:
The capability already exists in `subroutine lnfm_init` both in the implementation that we're working on now (in CNFireBaseMod.F90), as well as in my previous implementation (in SFMainMod.F90). All you need to do is change the variable to `active` because it is set to `default=inactive`.